packages/shared/src/hooks/feed/useFollowPostTags.ts

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useMemo } from 'react';
+import type { Post } from '../../graphql/posts';
+import useFeedSettings from '../useFeedSettings';
+import { featurePostTagSorting } from '../../lib/featureManagement';
+import { useConditionalFeature } from '../useConditionalFeature';
+import { useAuthContext } from '../../contexts/AuthContext';
+import useTagAndSource from '../useTagAndSource';
+import { Origin } from '../../lib/log';
+
+interface UseFollowPostTagsProps {
+  post: Post;
+}
+
+interface UseFollowPostTags {
+  isTagExperiment: boolean;
+  onFollowTag: (tag: string) => void;
+  tags: Record<'all' | 'followed' | 'notFollowed', string[]>;
+}
+
+export const useFollowPostTags = ({
+  post,
+}: UseFollowPostTagsProps): UseFollowPostTags => {
+  const { isLoggedIn } = useAuthContext();
+  const { feedSettings } = useFeedSettings();
```

Comment:
Don't think this now takes custom feeds into account right?

Reply:
I don't think this should take in account custom feeds since this is only shown in post page and not in feed. Since there is no way to detect if user come from a custom feed I'm assuming that you want to add tag to you main feed. Wdyt?

Reply:
Although for Feed we should also have this experiment so we might have to consider Custom Feed.

For feed, we should display the tags that the user follows only.

Reply:
Added custom feed support ✔️